examples/30_extended/blackbox_optimization_tutorial.py

```diff
@@ -0,0 +1,330 @@
+# a lambda function can be used in the BO function call to specify the extra
+# input parameters. We choose the latter here.
+
+
+def svm_wrapper(C, gamma, train_X, train_y, valid_X, valid_y):
+    clf = SVC(C=C, gamma=gamma)
+    clf.fit(train_X, train_y)
+    # BO maximizes accuracy
+    return accuracy_score(clf.predict(valid_X), valid_y)
+
+
+############################################################################
+# Running Bayesian Optimization
+# *****************************
+# We use the popular BO package `bayes_opt <https://github.com/fmfn/BayesianOptimization>`_.
+# We warm-start the BO loop with 3 random configurations to begin with.
+
+optimizer = BayesianOptimization(
+    f=lambda C, gamma: svm_wrapper(C, gamma, train_X, train_y, valid_X, valid_y),
+    pbounds=pbounds,
+    random_state=seed,
+)
+
+# Running BO optimization for 10 iterations after seeding model with 3 random evaluations
+# A total of 13 function evaluations
+optimizer.maximize(init_points=3, n_iter=10)
+
+# Best configuration
+best_config = optimizer.max["params"]
+print("Best found configuration: \n{}".format(best_config))
+print("Best accuracy obtained on validation set : {:.5f} %".format(optimizer.max["target"] * 100))
+
+############################################################################
+# Obtaining optimization trace
+# ****************************
+# Obtaining optimization trace and building an OpenMLRunTrace object for the run
+
+# Following these guidelines: https://docs.openml.org/OpenML_definition/#trace
+trace_iterations = dict()
+for i, _eval in enumerate(optimizer.res):
+    selected = False
+    if _eval == optimizer.max:
+        selected = True
+    trace_iterations[(0, 0, i)] = OpenMLTraceIteration(
+        repeat=0,
+        fold=0,
+        iteration=i,
+        setup_string="",
+        evaluation=_eval["target"],
+        selected=selected,
+        parameters=OrderedDict(_eval["params"]),
+    )
+run_trace = OpenMLRunTrace(-1, trace_iterations=trace_iterations)
+
+############################################################################
+# Refitting model with best configuration
+# ***************************************
+# Re-fitting model on complete training data and obtaining predictions locally
+
+model_params = dict(**best_config, random_state=seed, probability=True)
+
+clf = SVC(**model_params)
+clf.fit(train_X.append(valid_X), train_y.append(valid_y))
+_predictions = clf.predict(test_X)
+print("Accuracy on the test set: {:.5f} %".format(accuracy_score(_predictions, test_y) * 100))
+
+###########################################
+# Preparing an OpenML flow with refit model
+# *****************************************
+# Creating a new flow that will store the SVM model with hyperparameters
+# optimized for the chosen dataset.
+
+general = dict(
+    name="sklearn_bbo_example_flow",
+    description=("Running BO on SVM using OpenML"),
+    external_version="bayesian-optimization==1.2.0",
```

Comment:
Could you please add scikit-learn here, also can you use the installed version of `bayesian-optimization` as a dependency?

Reply:
> use the installed version of bayesian-optimization as a dependency? 

I didn't quite understand. `v.1.2.0` is the one that I have locally to create the example.

Reply:
What I meant is that you query the version via `bayes_opt.__version__` instead of putting a fixed version into the script.

Reply:
alright, understood and makes sense now
but how to handle this when `version` or `__version__` is not available from the package?